### PR TITLE
Add new package: py-pid

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pid/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    version('3.0.1', sha256='2f51b61210f8e1f009b09a2034717003ca22dcd86995537ecb857863bddca89a')
+    version('3.0.0', sha256='3d251eadedc6fbd1fe4b43d521e76b83afd244b8b1951a2cd96864406bc96381')
+    version('2.2.5', sha256='d4c68554bf4b2fc7d0b50749f535f5c1fceb74ff025ce1a3f06745d15c595d40')
+    version('2.2.4', sha256='de3cc35e18c5409d8424813ab422b637af4d25bfdcf2c15ee6c5af447778de22')
+    version('2.2.3', sha256='14555fc214e0dfee7d94598b759523349832597e163415d1a7b0d87d9902cc47')
+    version('2.2.2', sha256='716bb5803fed50facdb62be0e48d08dd95e7392fcfb03f5540915623f9c4ee44')
+    version('2.2.1', sha256='2c5b398d348b8b1901ccb29b5c914c583187692acfbc3c28fc4ee483b9909357')
+    version('2.2.0', sha256='f2c3beb5742159794379b73088eb3f592a4b7b93bfef95f8bbc27ab98e5394ed')
+
+    extends('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pid/package.py]: This line isn't needed, it's implied by PythonPackage. But the following line is needed:
```python
depends_on('py-setuptools', type='build')
```

File: var/spack/repos/builtin/packages/py-pid/package.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    depends_on('py-psutil', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pid/package.py]: According to the `setup.py`, this dependency is only required on Windows.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pid/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/py-pid/package.py]: Fixed in https://github.com/spack/spack/pull/16202/commits/a43e02cc2558ae97a2d7ee841b15cc77be824378

File: var/spack/repos/builtin/packages/py-pid/package.py (hunk 2 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/py-pid/package.py]: Fixed in https://github.com/spack/spack/pull/16202/commits/a43e02cc2558ae97a2d7ee841b15cc77be824378

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**